### ISSUE TITLE
error after registering 

### ISSUE THREAD

**Issue by leo1971** [gobuffalo/toodo#8]
`items/index.html: line 0: items/_li.html: line 1: item: unknown identifier
github.com/gobuffalo/pop.(*Connection).Transaction.func1
/Users/leonardoke/go/src/github.com/gobuffalo/pop/connection.go:131
github.com/gobuffalo/pop.(*postgresql).Lock
/Users/leonardoke/go/src/github.com/gobuffalo/pop/postgresql.go:164
github.com/gobuffalo/pop.(*Connection).Transaction
/Users/leonardoke/go/src/github.com/gobuffalo/pop/connection.go:118
github.com/gobuffalo/buffalo/middleware.glob..func1.1.1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/middleware/pop_transaction.go:23
github.com/gobuffalo/buffalo/middleware/csrf.glob..func1.1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/middleware/csrf/csrf.go:120
github.com/gobuffalo/buffalo/middleware.ParameterLogger.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/middleware/param_logger.go:28
github.com/gobuffalo/buffalo/middleware/ssl.ForceSSL.func1.1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/middleware/ssl/ssl.go:26
github.com/gobuffalo/buffalo.sessionSaver.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/session.go:70
github.com/gobuffalo/buffalo.RequestLoggerFunc.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/request_logger.go:51
github.com/gobuffalo/buffalo.(*App).PanicHandler.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/errors.go:71
github.com/gobuffalo/buffalo.RouteInfo.ServeHTTP
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/handler.go:80
github.com/gobuffalo/buffalo.(*RouteInfo).ServeHTTP
<autogenerated>:1
github.com/gorilla/mux.(*Router).ServeHTTP
/Users/leonardoke/go/src/github.com/gorilla/mux/mux.go:162
github.com/markbates/refresh/refresh/web.ErrorChecker.func1
/Users/leonardoke/go/src/github.com/markbates/refresh/refresh/web/web.go:23
net/http.HandlerFunc.ServeHTTP
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:1947
github.com/gobuffalo/buffalo.(*App).ServeHTTP
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/server.go:110
net/http.serverHandler.ServeHTTP
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:2694
net/http.(*conn).serve
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:1830
runtime.goexit
/usr/local/Cellar/go/1.10.1/libexec/src/runtime/asm_amd64.s:2361
github.com/gobuffalo/buffalo.sessionSaver.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/session.go:72
github.com/gobuffalo/buffalo.RequestLoggerFunc.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/request_logger.go:51
github.com/gobuffalo/buffalo.(*App).PanicHandler.func1
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/errors.go:71
github.com/gobuffalo/buffalo.RouteInfo.ServeHTTP
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/handler.go:80
github.com/gobuffalo/buffalo.(*RouteInfo).ServeHTTP
<autogenerated>:1
github.com/gorilla/mux.(*Router).ServeHTTP
/Users/leonardoke/go/src/github.com/gorilla/mux/mux.go:162
github.com/markbates/refresh/refresh/web.ErrorChecker.func1
/Users/leonardoke/go/src/github.com/markbates/refresh/refresh/web/web.go:23
net/http.HandlerFunc.ServeHTTP
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:1947
github.com/gobuffalo/buffalo.(*App).ServeHTTP
/Users/leonardoke/go/src/github.com/gobuffalo/buffalo/server.go:110
net/http.serverHandler.ServeHTTP
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:2694
net/http.(*conn).serve
/usr/local/Cellar/go/1.10.1/libexec/src/net/http/server.go:1830
runtime.goexit
/usr/local/Cellar/go/1.10.1/libexec/src/runtime/asm_amd64.s:2361
`

**Comment by houstonian** [comment]
I am encountering the same problem. I registered and added an item fine, then when I went to show list  I got the same output. Wondering what's going on. I'm looking into the templates.

**Comment by houstonian** [comment]
<%= for (item) in items { return partial("items/li.html") } %>

this line seems to be the problem. Not familiar enough with go and buffalo templates to find a solution.

**Comment by scoop206** [comment]
I hit this too.  
The problem was that when showing the list of items, the 'item' variable wasn't in the context that was available when the _li.html partial was being rendered.

It was also hard to get to the problem. By this I mean that I would modify the template that was calling the partial, and I wasn't seeing a change when reloading the webpage.
I beat my head against it for too long and then concluded that the app.Resource("/items", ItemsResource{}) in actions/app.go must be generating the templates already. So that it's basically ignoring what's on disk in the templates folder. (I'm not positive on this aspect but pushed forward on that hunch)

I cracked the buffalo source and could see in the generators that the current buffalo CRUD templates were newer. They now take a different approach. Rather than using a for loop to call partial for each they just do them all in one page. So the whole 'item' not in the context is no longer an issue if the app were to be rebuilt.

So that's what I did along with haphazardly documenting my experience in the README.
https://github.com/scoop206/buffalo-todo2
